src/Illuminate/Container/Container.php

```diff
@@ -234,6 +236,10 @@ public function bind($abstract, $concrete = null, $shared = false)
         // bound into this container to the abstract type and we will just wrap it
         // up inside its own Closure to give us more convenience when extending.
         if (! $concrete instanceof Closure) {
+            if (! is_string($concrete)) {
```

Comment:
`! is_string($concrete) && ! is_null($concrete)`

Reply:
the additional `is_null` check here would only be beneficial if `$abstract` is also null. Do you want me to protect against that?

Reply:
No, I think this is good enough. :)